src/topology/separation.lean

```diff
@@ -16,6 +16,58 @@ variables {α : Type u} {β : Type v} [topological_space α]
 
 section separation
 
+/--
+`separate` is a predicate on pairs of `finset`s of a topological space.  It holds if the two
+`finset`s are contained in disjoint open sets.
+-/
+def separate : finset α → finset α → Prop :=
+  λ (s t : finset α), ∃ U V : (set α), (is_open U) ∧ is_open V ∧
+  (∀ a : α, a ∈ s → a ∈ U) ∧ (∀ a : α, a ∈ t → a ∈ V) ∧ disjoint U V
+
+@[symm] lemma separate.symm {s t : finset α} : separate s t → separate t s :=
+begin
+  rintros ⟨U, V, oU, oV, aU, bV, UV⟩,
+  exact ⟨V, U, oV, oU, bV, aU, disjoint.symm UV⟩
+end
+
+lemma separate.comm (s t : finset α) : separate s t ↔ separate t s :=
+⟨separate.symm, separate.symm⟩
+
+lemma separate.empty_right : ∀ {a : finset α}, separate a ∅ :=
+λ a, ⟨_, _, is_open_univ, is_open_empty, λ a h, mem_univ a, λ a h, by cases h, disjoint_empty _⟩
+
+lemma separate.empty_left : ∀ {a : finset α}, separate ∅ a :=
+λ _, separate.empty_right.symm
```

Comment:
Idem

Reply:
So this would become `(separate.empty_right a).symm`, if you make the argument explicit.